tests/components/permobil/test_config_flow.py

```diff
@@ -105,6 +109,47 @@ async def test_config_flow_incorrect_code(
     assert result["errors"]["base"] == "invalid_code"
 
 
+async def test_config_flow_unsigned_eula(
+    hass: HomeAssistant, my_permobil: Mock
+) -> None:
+    """Test the config flow from start to until email code verification and the user has not accepted the eula."""
+    my_permobil.request_application_token.side_effect = MyPermobilEulaException
+    # init flow
+    with patch(
+        "homeassistant.components.permobil.config_flow.MyPermobil",
+        return_value=my_permobil,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={CONF_EMAIL: MOCK_EMAIL},
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "region"
+    assert result["errors"] == {}
+
+    # select region step
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_REGION: MOCK_REGION_NAME},
+    )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "email_code"
+    assert result["errors"] == {}
+
+    # request region code
+    # here the request_application_token raises a MyPermobilEulaException
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_CODE: MOCK_CODE},
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "email_code"
+    assert result["errors"]["base"] == "unsigned_eula"
```

Comment:
After this, can we make the test retry the submission and patch the lib properly to test if the flow is able to succeed?